fallen/DDEngine/Source/figure.cpp

```
@@ -2727,6 +2727,16 @@ void FIGURE_draw_prim_tween(
 
     LOG_ENTER(Figure_Draw_Prim_Tween)
 
+    ASSERT(p_thing)
+
+    if (!p_thing)
+        return;
```

whats the `assert` for if you are gonna return anyway?

---

For debugging purposes. An invalid level/figure entity will cause the assert in the debug builds.

---

but you are gonna return anyway so thats not an error.

---

It _is_ an error from the standpoint of the game. Only the beta/old level formats cause this to happen.

The structure around this function doesn't allow for any real error handling without doing major rewrites, hence why it returns immediately. The returns are there in place to avoid crashes.

If you don't want the asserts, I'll remove them. But they _should_ be treated as errors.

---

If they should be treated as errors then they should not be silently ignored. Its either one or the other, otherwise the code doesnt make sense

---

There isn't really a good solution to this issue without doing a rewrite of this part of the code (either by the means of exceptions or return error codes).

The only other thing I could do is make it throw an explicit error message box no matter the type of build it is, but that isn't a particularly good user experience.

Either way, I've removed the asserts in https://github.com/TracerDS/UrbanChaos/pull/15/commits/f803228bdb7afb93ee706343d14a225b02d9b157

Also, they're not silent if they assert in a Debug build. They're not critical errors, mind you. I could add it to the log, I suppose. But that's about it.